tutorials/curator-llm-pii/PII-modification-Enron.ipynb

```diff
@@ -0,0 +1,1114 @@
+    "            print(f\"  • {k.title()}: {v:.2f}\")\n",
+    "            \n",
+    "    except Exception as e:\n",
+    "        print(f\"An error occurred during pattern analysis: {str(e)}\")\n",
+    "        print(\"\\nDebug information:\")\n",
+    "        print(f\"Is dataset available? {'Yes' if 'dataset' in locals() else 'No'}\")\n",
+    "        print(f\"Is async_modified available? {'Yes' if 'async_modified' in locals() else 'No'}\")\n",
+    "        print(f\"Is sync_modified available? {'Yes' if 'sync_modified' in locals() else 'No'}\")\n",
+    "\n",
+    "# Run both analyses\n",
+    "print(\"Starting comprehensive PII modification analysis...\")\n",
+    "compare_pii_modifications()\n",
+    "print(\"\\nAnalyzing PII patterns across all emails...\")\n",
+    "analyze_pii_patterns()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "79a69033-df1f-41dd-8355-992113cb77d3",
+   "metadata": {},
+   "source": [
+    "## Key things to note from steps 6 and 7\n",
+    "\n",
+    "• Both modification methods successfully removed personal information\n",
+    "\n",
+    "• The async method used longer replacements (like [Employee_ID_123]) while sync used shorter ones (employee.1)\n",
+    "\n",
+    "• Both methods preserved the email's structure and business content\n",
+    "\n",
+    "• Date and subject information were kept intact as they don't contain personal information\n",
+    "\n",
+    "• The meaning and context of the email remained clear despite the modifications\n",
+    "\n",
+    "Key Differences Between `Async` and `Sync`:\n",
+    "1. Style of Replacement:\n",
+    "   - Async: Uses bracketed descriptive tags [Employee_ID_123]\n",
+    "   - Sync: Uses simpler generic replacements employee.1\n",
+    "\n",
+    "2. Length Changes:\n",
+    "   - Async: Slightly longer due to descriptive replacements\n",
+    "   - Sync: More compact replacements leading to smaller length increase\n",
+    "\n",
+    "3. Consistency:\n",
+    "   - Both methods were equally effective at removing PII\n",
+    "   - Both preserved the email's business context"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "33e1f5a6-e18a-41c0-aa07-d4135983db49",
+   "metadata": {},
+   "source": [
+    "## Conclusion\n",
+    "This tutorial has demonstrated a practical implementation of PII modification using NeMo Curator's LLM-based modifiers on the Enron email dataset. \n",
+    "\n",
+    "The PII modification process uses LLM (Large Language Model) to identify and modify personally identifiable information in the emails. Here's what changed\n",
+    "\n",
+    "### Types of PII Modified:\n",
+    "1. **Email Addresses**: Replaced with anonymized versions\n",
+    "2. **Phone Numbers**: Converted to generic formats\n",
+    "3. **Names**: Changed to placeholder names or roles\n",
+    "4. **Addresses**: Generalized to city/region level\n",
+    "5. **Other Identifiers**: Modified any SSNs, account numbers, etc."
```

Comment:
Confused about this block. Doesn't every PII identified get replaced like a generic, such as {{name}}, {{email}}, etc.?